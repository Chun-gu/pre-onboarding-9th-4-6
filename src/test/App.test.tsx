import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import '@testing-library/jest-dom';

import { OrderTable } from '@/components';

const queryClient = new QueryClient();

describe(`<OrderTable />`, () => {
  it('matches snapshot', () => {
    const utils = render(
      <QueryClientProvider client={queryClient}>
        <BrowserRouter>
          <OrderTable />
        </BrowserRouter>
      </QueryClientProvider>,
    );

    expect(utils.container).toMatchSnapshot();
  });
});

describe(`<OrderTable />`, () => {
  it('select match', () => {
    const utils = render(
      <select data-testid="select">
        <option value="">전체</option>
        <option value="true">O</option>
        <option value="false">X</option>
      </select>,
    );
    expect(utils.container).toMatchSnapshot();
  });
});
