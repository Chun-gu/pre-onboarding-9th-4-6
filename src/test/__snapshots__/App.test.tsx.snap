// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Counter/> > mathch 1`] = `
<div>
  <div>
    Loading...
  </div>
</div>
`;

exports[`<OrderTable /> > matches snapshot 1`] = `
<div>
  <div>
    Loading...
  </div>
</div>
`;

exports[`<OrderTable /> > select match 1`] = `
<div>
  <select
    data-testid="select"
  >
    <option
      value=""
    >
      전체
    </option>
    <option
      value="true"
    >
      O
    </option>
    <option
      value="false"
    >
      X
    </option>
  </select>
</div>
`;
